src/zarr/codecs/bytes.py

```diff
@@ -25,7 +25,7 @@ class Endian(Enum):
 
 
 @dataclass(frozen=True)
-class BytesCodec(ArrayBytesCodec):
+class BytesCodec(ArrayBytesCodecBatchMixin):
```

Comment:
Will `ArrayBytesCodec` be used anywhere other than as the base class fro `ArrayBytesCodecBatchMixin`? If not, we might want to consider simplifying the inheritance structure a bit.

Reply:
I would anticipate that folks want to build a codec that implements their own batching, e.g. in rust or on the GPU. That is why, we should keep both classes.